pkg/infra/iac2/templates_compiler.go

```diff
@@ -877,6 +882,30 @@ func (tc TemplatesCompiler) associateRouteTable(out io.Writer, rt *resources.Rou
 	return errs.ErrOrNil()
 }
 
+func (tc TemplatesCompiler) attachToTargetGroup(out io.Writer, tg *resources.TargetGroup) error {
+	var errs multierr.Error
+
+	_, err := out.Write([]byte("\n\n"))
+	errs.Append(err)
+
+	for _, target := range tg.Targets {
+
+		attachment := &TargetGroupAttachment{
```

Comment:
so i added the Targets field to target group field since thats how its represented truly in aws. aws native for pulumi now goes that route as well. 

The reason i think this is ok is because we still have the information in the provider and nothing is derived in the iac plugin for whats being setup, it just represents it in a different way.

Im ok to change it if you really want, but the targets field feels way cleaner to me and is in better parity with the aws sdks and apis

Reply:
I suppose it depends on whether we want the graph to represent them "as they are, once established" versus "as they are created". While there's a convenience way to hook the targets in, there must be the Attachment route to resolve circular dependencies (Target -> Unit -> LB + LB -> Target). The former makes it easier to mentally match the graph to the console, but the latter makes it easier to implement IaC plugins (since this is common logic).

I'm fine with either approach, as long as we are making the conscious decision.